src/main/java/core/basesyntax/FIgureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FIgureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(color,
+                        random.nextDouble() * 10 + 1);
+            case 1:
+                return new Rectangle(color,
+                        random.nextDouble() * 10 + 1,
+                        random.nextDouble() * 10 + 1);
```

Comment:
For 'Rectangle', separate variables should be created for the width and height before passing them to the constructor to improve code readability.